### PR TITLE
Issue #13345: Enable examples tests for IllegalCatchCheck

### PR DESCRIPTION
**Pull request by AmitKumarDeoghoria**
### Title: Enable examples tests for  IllegalCatchCheck

**Related Issue: #13345** 

**Summary**
This pull request aims to enable example tests for the ** IllegalCatchCheck** in the project. The implementation focuses on ensuring that the  IllegalCatch checks are correctly validated through comprehensive test cases, improving overall code reliability and quality.

**Key Changes Made**

- Conversion of txt file to Java File
- Enable the test cases by adding expected violations in the expected array for all example tests
- Made corresponding changes in the xml and xml.template file

### CODE REVIEW

**Comment by AmitKumarDeoghoria** [comment]
GitHub, generate website

**Comment by romani** [comment]
Please align content of java code https://www.diffchecker.com/5X7HJcvG/

If violation is gone, just keep empty line.

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/coding/illegalcatch/Example2.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
+/*xml
+<module name="Checker">
+  <module name="TreeWalker">
+    <module name="IllegalCatch">
+      <property name="illegalClassNames" value="ArithmeticException,
```
Comment (romani) [inline src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/coding/illegalcatch/Example2.java:5]: Please put on new line `value=`

**Review by romani — CHANGES_REQUESTED** [review]
Items

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/coding/illegalcatch/Example2.java (hunk 1 above)
Comment (AmitKumarDeoghoria) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/coding/illegalcatch/Example2.java]: Done.

**Comment by AmitKumarDeoghoria** [comment]
GitHub, generate website

**Review by romani — APPROVED** [review]
Awesome https://www.diffchecker.com/FHr5tiqc/

**Pull request merged**